internal/files/sign.go

```diff
@@ -0,0 +1,110 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package files
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/ProtonMail/gopenpgp/v2/armor"
+	"github.com/ProtonMail/gopenpgp/v2/constants"
+	"github.com/ProtonMail/gopenpgp/v2/crypto"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/logger"
+)
+
+const (
+	signerPrivateKeyfileEnv = "ELASTIC_PACKAGE_SIGNER_PRIVATE_KEYFILE"
+	signerPassphraseEnv     = "ELASTIC_PACKAGE_SIGNER_PASSPHRASE"
+
+	signatureComment = "Signed with elastic-package (using GopenPGP: https://gopenpgp.org)"
+)
+
+type SignOptions struct {
+	PackageName    string
+	PackageVersion string
+}
+
+// VerifySignerConfiguration function verifies if the signer configuration is complete.
+func VerifySignerConfiguration() error {
+	signerPrivateKeyfile := os.Getenv(signerPrivateKeyfileEnv)
+	if signerPrivateKeyfile == "" {
+		return fmt.Errorf("signer private keyfile is required. Please define it with environment variable %s", signerPrivateKeyfileEnv)
+	}
+
+	_, err := os.Stat(signerPrivateKeyfile)
+	if err != nil {
+		return errors.Wrap(err, "can't access the signer private keyfile")
+	}
+
+	signerPassphrase := os.Getenv(signerPassphraseEnv)
```

Comment:
When supporting community contributors we may need other sources of passphrases, as the OS keyrings or asking them in the CLI. It can be risky to keep the password in an environment variable for long.
We may also consider supporting Vault, so keys are obtained directly from there.
But we are good by now with this method.

Reply:
Good ideas but as candidates for further improvement. Let's deliver the first few iterations of the solution.